contracts/andromeda_vault/src/contract.rs

```diff
@@ -0,0 +1,426 @@
+use ado_base::state::ADOContract;
+use andromeda_protocol::vault::{
+    ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg, StrategyAddressResponse, StrategyType,
+    BALANCES, STRATEGY_CONTRACT_ADDRESSES,
+};
+use common::{
+    ado_base::{
+        recipient::Recipient, AndromedaMsg, AndromedaQuery, InstantiateMsg as BaseInstantiateMsg,
+        QueryMsg as AndrQueryMsg,
+    },
+    error::ContractError,
+    parse_message, require,
+    withdraw::{Withdrawal, WithdrawalType},
+};
+use cosmwasm_std::{
+    coin, entry_point, to_binary, BankMsg, Binary, Coin, ContractResult, CosmosMsg, Deps, DepsMut,
+    Empty, Env, MessageInfo, Order, QueryRequest, ReplyOn, Response, SubMsg, SystemResult, Uint128,
+    WasmMsg, WasmQuery,
+};
+use cw2::{get_contract_version, set_contract_version};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:andromeda-rates";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[entry_point]
```

Comment:
nit: Replace all `#[entry_point]` tags with `#[cfg_attr(not(feature = "library"), entry_point)]`